src/internal/core.rs

```diff
@@ -130,6 +123,17 @@ impl<P: Package, V: Version> State<P, V> {
                     _ => {}
                 }
             }
+            if let Some(incompat_id) = conflict_id {
+                let (package_almost, root_cause) = self.conflict_resolution(incompat_id)?;
+                self.unit_propagation_buffer.clear();
+                self.unit_propagation_buffer.push(package_almost.clone());
```

Comment:
I originally added a `self.unit_propagation_buffer.push(current_package);`, after this line to have it look to see if there are other incompatibilities that are still relevant related to `current_package`. That would better match what the old code did. But all tests passed without that line, including comparing the `elm` file to the sat solver, so it seams not to be needed.